contrib/epee/src/net_helper.cpp

```diff
@@ -78,6 +118,85 @@ namespace net_utils
 		});
 		return shared->result_.get_future();
 	}
+
+	boost::system::error_code blocked_mode_client::try_connect(const std::string& addr, const std::string& port, std::chrono::milliseconds timeout)
+	{
+		m_deadline.expires_from_now(timeout);
+		boost::unique_future<boost::asio::ip::tcp::socket> connection = m_connector(addr, port, m_deadline);
+		for (;;)
+		{
+			m_io_service.reset();
+			m_io_service.run_one();
+
+			if (connection.is_ready())
+				break;
+		}
+
+		m_ssl_socket->next_layer() = connection.get();
+		m_deadline.cancel();
+		if (m_ssl_socket->next_layer().is_open())
+		{
+			boost::system::error_code error{};
+			m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
+			// SSL Options
+			if (m_ssl_options.support == ssl_support_t::e_ssl_support_enabled || m_ssl_options.support == ssl_support_t::e_ssl_support_autodetect)
+			{
+				if ((error = m_ssl_options.handshake(*m_ssl_socket, boost::asio::ssl::stream_base::client, addr, timeout)))
+				{
+					MWARNING("Failed to establish SSL connection");
+					boost::system::error_code ignored_ec;
+					m_ssl_socket->next_layer().shutdown(boost::asio::ip::tcp::socket::shutdown_both, ignored_ec);
+					m_ssl_socket->next_layer().close();
```

Comment:
Do you need to clear m_stream_error here, if not used only by connect ?

Reply:
I do not believe `m_stream_error` needs to be reset here. `try_connect` is called from `connect` which always resets the value based on the results from `try_connect`.